props/jetpack_Oli_buttons.h

```diff
@@ -0,0 +1,931 @@
+      return;
+    }
+    track_num_ = (track_num_ % num_tracks_) + 1; // Cycle to the next track
+    PlayTrack();
+  }
+
+  void PreviousTrack() {
+    if (num_tracks_ <= 0) {
+      PVLOG_DEBUG << "No tracks available.\n";
+      return;
+    }
+    track_num_ = (track_num_ - 2 + num_tracks_) % num_tracks_ + 1; // Cycle to the previous track
+    PlayTrack();
+  }
+
+// only one prop can have chdir & Parse, so in a dual_prop.h/multi_prop.h "environment" SaberFett263Button already has them.
+#ifndef PROPS_SABER_FETT263_BUTTONS_H
+  bool chdir(const StringPiece dir) override {
+    bool ret = PropBase::chdir(dir);
+    num_tracks_ = RunCommandAndGetSingleLine("list_current_tracks", nullptr, 0, 0, 0);
+    track_num_ = 0;
+    return ret;
+  }
+
+  bool Parse(const char *cmd, const char* arg) override {
+    if (PropBase::Parse(cmd, arg)) return true;
+    if (!strcmp(cmd, "list_current_tracks")) {
+    //PVLOG_NORMAL << "Tracks available: " << num_tracks_ << " (including default: " << num_default_tracks_ << ")\n";
+      LOCK_SD(true);
+      for (const char* dir = current_directory; dir; dir = next_current_directory(dir)) {
+        PathHelper path(dir, "tracks");
+        ListTracks(path);
+      }
+      LOCK_SD(false);
+      return true;
+    }
+    if (!strcmp(cmd, "list_current_tracks")) {
+      // Tracks must be in: font/tracks/*.wav
+      LOCK_SD(true);
+      for (const char* dir = current_directory; dir; dir = next_current_directory(dir)) {
+        PathHelper path(dir, "tracks");
+        ListTracks(path);
+      }
+      LOCK_SD(false);
+      return true;
+    }
+    return false;
+  }
+#endif // PROPS_SABER_FETT263_BUTTONS_H
+#endif // JETPACK_TRACK_PLAYER
+
+  // I used SA22C Event2 as a base, and I added my "twist" on it, since a jetpack prop/backpack with a
+  // Proffie board in it can't realisticly have twist or stab motion events.
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, modifiers)) {
+#ifdef BLADE_DETECT_PIN
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_OFF):
+        blade_detected_ = true;
+        FindBladeAgain();
+        SaberBase::DoBladeDetect(true);
+        return true;
+
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_OFF, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_OFF, MODE_ANY_BUTTON | MODE_OFF):
+        blade_detected_ = false;
+        FindBladeAgain();
+        SaberBase::DoBladeDetect(false);
+        return true;
+#endif // BLADE_DETECT_PIN
+      // Jetpack ON AND Volume Up:
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_ON):
+        if (!mode_volume_) {
+          if       (idle_) StartFlightMode(); // Transition from idle to flight  (rev more up)

```

Comment:
This is not how you make an if statement easy to read, please remove extra spaces and put the function call on the next line.